### PR TITLE
Update some commentary.

### PR DESCRIPTION
**Pull request by bangerth**
Follow-up to #8341.

@marcfehling -- FYI.

### CODE REVIEW

File: include/deal.II/hp/refinement.h [hunk 1]
```diff
@@ -293,28 +293,44 @@ namespace hp
...
+     * coarsening. This assumption is often not satisfied in practice:
+     * For example, if a cell is at a corner singularity, then the one
+     * child cell that ends up closest to the singularity will inherit
+     * the majority of the remaining error -- but this function can
+     * not know where the singularity will be, and consequently
+     * assumes equal distribution.
+     *
+     * When transferring the predicted error to the coarsened mesh,
+     * make sure to configure your CellDataTransfer object with
+     * GridTools::CoarseningStrategies::sum() as a coarsening
```
Comment (marcfehling) [inline include/deal.II/hp/refinement.h]: We moved the `CoarseningStrategies` namespace out of the `GridTools` one. I missed to adapt this part of the documentation at some point...
```suggestion
     * CoarseningStrategies::sum() as a coarsening
```

File: include/deal.II/hp/refinement.h [hunk 2]
```diff
@@ -293,28 +293,44 @@ namespace hp
...
-     * quadratic solution), we are confident to say that the error will not
-     * change by sole interpolation on the larger finite element space.
+     * finite element on cell $K$. Here, we assume that the finite element will
+     * not change in the adaptation process so that $p_K = \text{const}$.
+     * However during coarsening, the finite elements on siblings may be
+     * different, and their parent cell will be assigned to their least
+     * dominating finite element that belongs to its most general child. Thus,
+     * we will always interpolate on an enclosing finite element space.
+     * Additionaly assuming that the finite elements on the cells to be
+     * coarsened are sufficient to represent the solution correct (e.g. at least
```
Comment (kronbichler) [inline include/deal.II/hp/refinement.h]:
```suggestion
     * coarsened are sufficient to represent the solution correctly (e.g. at least
```

**Comment by bangerth** [comment]
I made the suggested changes.

**Comment by masterleinad** [comment]
You need to fix the indentation.

**Comment by bangerth** [comment]
Ah, I was under the impression that we no longer indent comments. Oh well. I'll get to it tonight when I'm back at my laptop!

**Comment by bangerth** [comment]
Indentation fixed.

**Pull request merged**